packages/internal/src/natural-sort.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @param {string} a
+ * @param {string} b
+ * @returns {-1 | 0 | 1}
+ */
+const compareNats = (a, b) => {
```

Comment:
My understanding is that this logic works even is the params are `number`, right?

Reply:
Mostly, but not when either is NaN or both are infinite. And this function is not exported, so we can count on the input arguments always being decimal digit strings.